### PR TITLE
Add tag / build reference api

### PR DESCRIPTION
**Pull request by filippomc**
Closes [CH-47](https://metacell.atlassian.net/browse/CH-47)

Implemented solution: the build id (as git commit hash) is now extracted by harness-deployment and added to values.yaml.
The common service exposes it and the tag.
Added an example on samples microservice
image

How to test this PR: start samples.DOMAIN and verify that is shows the tag and build id as in the screenshot.

### Sanity checks:
- [ ] The pull request is explicitly linked to the relevant issue(s)
- [ ] The issue is well described: clearly states the problem and the general proposed solution(s)
- [ ] From the issue and the current PR it is explicitly stated how to test the current change
- [ ] The labels in the issue set the scope and the type of issue (bug, feature, etc.)
- [ ] All the automated test checks are passing
- [ ] All the linked issues are included in one milestone
- [ ] All the linked issues are in the Review/QA column of the [board](https://app.zenhub.com/workspaces/cloud-harness-5fdb203b7e195b0015a273d7/board)
- [ ] All the linked issues are assigned

### Breaking changes (select one):
- [ ] The present changes do not change the preexisting api in any way
- [ ] This PR and the issue are tagged as a `breaking-change`

### Possible deployment updates issues (select one):
- [ ] There is no reason why deployments based on CloudHarness may break after the current update
- [ ] This PR and the issue are tagged as `alert:deployment`

### Test coverage (select one):
- [ ] Tests for the relevant cases are included in this pr
- [ ] The changes included in this pr are out of the current test coverage scope

### Documentation (select one):
- [ ] The documentation has been updated to match the current changes
- [ ] The changes included in this PR are out of the current documentation scope

### Nice to have (if relevant):
- [ ] Screenshots of the changes
- [ ] Explanatory video/animated gif

[CH-47]: https://metacell.atlassian.net/browse/CH-47?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Review by zsinnema — COMMENTED** [review]
question: @filippomc isn't it the case that for CH all apps in the deployment have the same version? Then imo we only need to implement this only in one application / endpoint (f.e. common). Instead of in all apps 

or may be in the keycload login template then also apps like jupyter hub are covered

**Comment by filippomc** [comment]
@zsinnema the version of this api is representative of the deployment so it's the same for all application. It is indeed implemented only for the common app, what the samples app does here is to display the value in the ui.

**Comment by zsinnema** [comment]
> @zsinnema the version of this api is representative of the deployment so it's the same for all applications. It is indeed implemented only for the common app, what the samples app does here is to display the value in the ui.

what I meant @filippomc is that we don't need an application to show the version number. We can just show it in the keycloak login screen.....

File: applications/samples/frontend/src/components/Version.tsx [hunk 1]
```diff
@@ -0,0 +1,19 @@
+import React, { useState, useEffect } from 'react';
+
+
+
+const Version = () => {
+  const [result, setResult] = useState(null);
+  useEffect(() => {
```
Comment (zsinnema) [inline applications/samples/frontend/src/components/Version.tsx:7]: @filippomc please move this to the keycloak login page template 
imo we should show the version there
and also we could show it on the keycloack account details page (the one of the user, where you can go to for f.e. changing your password/details)

**Review by zsinnema — CHANGES_REQUESTED** [review]

**Comment by filippomc** [comment]
@zsinnema I think it can be a good idea to show the version in the keycloak login screen, but we are not customizing the keycloak theme on ch, which makes updates simpler. Also derived applications have their theme and might not want to show the version there.

**Review by zsinnema — APPROVED** [review]

**Pull request merged**